### PR TITLE
crimson/tools: support more functions and rename

### PR DESCRIPTION
**Pull request by aisakaki**
Support iostat analysis.

Support Fio RBD rand read and Fio RBD rand write base cases.

Support collecting cpu frequency.

Support booting with a single core.

Merge two thread interfaces into one, Task.

change the name of rados_bench_tools

### CODE REVIEW

File: tools/crimson/crimson_stress_tool.py [hunk 1]
```diff
@@ -35,46 +34,23 @@ def run(self):
...
-        pass
-    @staticmethod
-    def post_process(env, test_case_result):
-        pass
-
-
-class ExecRadosBenchThread(ITestCaseBasedThread):
-    def __init__(self, thread_num, env):
-        ITestCaseBasedThread.__init__(self, thread_num, env)
+        Task.__init__(self, env)
```
Comment (tchaikov) [inline tools/crimson/crimson_stress_tool.py:53]:
```suggestion
        super().__init__(env)
```

File: tools/crimson/crimson_stress_tool.py [hunk 2]
```diff
@@ -138,11 +114,99 @@ def post_process(env, test_case_result):
...
+                        result_dic[self.lat] = float(match_res[4:-1]) #usec
+                if temp_lis[0] == "bw":
+                    match_res = re.search(r'avg=.*?,', line).group()
+                    result_dic[self.bw] = float(match_res[4:-1]) #kB/s
+                if temp_lis[0] == "iops":
+                    match_res = re.search(r'avg=.*?,', line).group()
+                    result_dic[self.iops] = float(match_res[4:-1]) 
+            line = self.result.readline()
+        self.result.close()
+        return result_dic
```
Comment (tchaikov) [inline tools/crimson/crimson_stress_tool.py:169]: might want add an empty line between methods.

Comment (aisakaki) [reply tools/crimson/crimson_stress_tool.py:187]: ok, I added an empty line before every methods.

File: tools/crimson/crimson_stress_tool.py (hunk 1 above)
Comment (aisakaki) [reply tools/crimson/crimson_stress_tool.py]: modified in new push.

File: tools/crimson/rados_bench_tools.py [hunk 3]
```diff
@@ -138,11 +114,97 @@ def post_process(env, test_case_result):
...
-class ReactorUtilizationCollectorThread(ITimePointBasedThread):
+
+class FioRBDRandWriteThread(Task):
     def __init__(self, env):
-        ITimePointBasedThread.__init__(self, env)
+        Task.__init__(self, env)
+        self.rw = "randwrite"
+        self.io_depth = 2  
+        self.io_engine = "rbd"
+        self.num_job = env.thread_num
```
[the remark this answers is not in the document]
Comment (aisakaki) [reply tools/crimson/rados_bench_tools.py]: corrected~

File: tools/crimson/crimson_stress_tool.py [hunk 4]
```diff
@@ -109,31 +121,38 @@ def pre_process(env):
...
     def __init__(self, env):
-        Task.__init__(self, env)
+        super().__init__(env)
+        self.task_set = env.args.taskset
         self.rw = "randwrite"
-        self.io_depth = 2  
+        self.io_depth = env.thread_num  
         self.io_engine = "rbd"
-        self.num_job = env.thread_num 
+        self.num_job = 1
```
Comment (cyx1231st) [inline tools/crimson/crimson_stress_tool.py:141]: It is better to merge this change (and others) into the previous commit. The commit title "change the name of rados_bench_tools" doesn't match the content.

**Review by cyx1231st — APPROVED** [review]
LGTM

File: tools/crimson/crimson_stress_tool.py (hunk 4 above)
Comment (aisakaki) [reply tools/crimson/crimson_stress_tool.py:141]: I separated unrelated changes from the latest commit in new push.

**Pull request merged**